lib/ansible/utils/ssh_agent.py

```diff
@@ -0,0 +1,666 @@
+        self._sock.connect(auth_sock)
+
+    def close(self):
+        self._sock.close()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+    def send(self, msg: bytes) -> bytes:
+        length = uint32(len(msg)).to_blob()
+        self._sock.sendall(length + msg)
+        bufsize = uint32.from_blob(self._sock.recv(4))
+        resp = self._sock.recv(bufsize)
+        if resp[0] == ProtocolMsgNumbers.SSH_AGENT_FAILURE:
+            raise SshAgentFailure('agent: failure')
+        return resp
+
+    def remove_all(self):
+        self.send(
+            ProtocolMsgNumbers.SSH_AGENTC_REMOVE_ALL_IDENTITIES.to_blob()
+        )
+
+    def remove(self, public_key: CryptoPublicKey):
+        key_blob = PublicKeyMsg.from_public_key(public_key).to_blob()
+        self.send(
+            ProtocolMsgNumbers.SSH_AGENTC_REMOVE_IDENTITY.to_blob() +
+            uint32(len(key_blob)).to_blob() + key_blob
+        )
+
+    def add(
+            self,
+            private_key: CryptoPrivateKey,
+            comments: str | None = None,
+            lifetime: int | None = None,
+            confirm: bool | None = None,
+    ):
+        key_msg = PrivateKeyMsg.from_private_key(private_key)
+        key_msg.comments = unicode_string(comments or '')
+        if lifetime:
+            key_msg.constraints += constraints(
+                [ProtocolMsgNumbers.SSH_AGENT_CONSTRAIN_LIFETIME]
+            ).to_blob() + uint32(lifetime).to_blob()
+        if confirm:
+            key_msg.constraints += constraints(
+                [ProtocolMsgNumbers.SSH_AGENT_CONSTRAIN_CONFIRM]
+            ).to_blob()
+
+        if key_msg.constraints:
+            msg = ProtocolMsgNumbers.SSH_AGENTC_ADD_ID_CONSTRAINED.to_blob()
+        else:
+            msg = ProtocolMsgNumbers.SSH_AGENTC_ADD_IDENTITY.to_blob()
+        msg += key_msg.to_blob()
+        self.send(msg)
+
+    def list(self) -> KeyList:
+        req = ProtocolMsgNumbers.SSH_AGENTC_REQUEST_IDENTITIES.to_blob()
+        r = memoryview(bytearray(self.send(req)))
+        if r[0] != ProtocolMsgNumbers.SSH_AGENT_IDENTITIES_ANSWER:
+            raise SshAgentFailure(
+                'agent: non-identities answer received for identities list'
+            )
+        return KeyList.from_blob(r[1:])
+
+    def lock(self, passphrase: bytes):
+        self.send(
+            ProtocolMsgNumbers.SSH_AGENTC_LOCK.to_blob() + AgentLockMsg(binary_string(passphrase)).to_blob()
+        )
+
+    def unlock(self, passphrase: bytes):
+        self.send(
+            ProtocolMsgNumbers.SSH_AGENTC_UNLOCK.to_blob() + AgentLockMsg(binary_string(passphrase)).to_blob()
+        )
+
+    def __contains__(self, public_key: CryptoPublicKey) -> bool:
+        msg = PublicKeyMsg.from_public_key(public_key)
+        return msg in self.list().keys
+
+
+def load_private_key(key_data: bytes, passphrase: bytes) -> CryptoPrivateKey:
```

Comment:
The actual return type is this union:
```suggestion
def load_private_key(key_data: bytes, passphrase: bytes) -> ssh.SSHPrivateKeyTypes | cryptography.hazmat.primitives.asymmetric.types.PrivateKeyTypes:
```

*(it's not imported, so I've just added the full path as an example, you'll need to import things)*